### DISCUSSION TITLE
Is it considered safe/supported to set the RetroNAS top level storage directory to one that's already been setup/populated by RetroNAS?

### DISCUSSION THREAD

**Discussion by jamescadd** [retronas/discussions#71] [Q&A]
My use case is I had RetroNAS (as a VM on ProxMox) up and running fine with the storage directory backed by an NFS mount on TrueNAS.  Loaded up a bunch of PS2 games and a few other systems and things were good.  Long story short I shot myself in the foot with fdisk and had to reinstall RetroNAS from scratch (yes there are many problems with that statement, please don't judge :).

I'm wondering if RetroNAS will generally be non-destructive in its approach to the data in the storage folder, now & I hope going forward.  My next steps will be to re-install PS2 and several other shares, and it would be a bummer if RetroNAS wiped out stuff I already had setup.

**Comment by sairuk** [comment]
I know @danmons covered this in detail somewhere but I can't see it in the wiki, we should add something

tl;dr Yes its is intended to be safe, but please read all below.

Ansible will not overwrite/delete data since we don't force changes. It will first check for the existence of 'something' and its state then if it is not in as defined in the playbook it _may_ fix it (if its safe to do so) or it will just report it.

The only thing you may run into is changes between versions, if we were to change some symlinks the data may not present to the service in exactly the same way but we use symlinks intentionally for this flexibility, the data is intended to remain untouched.

Even if major changes come down the line some day we'd either have to write a separate migration script/playbook _or_ communicate the change process because Ansible is that hands off in the approach we've taken.

If you have any concerns my advice would be to test with some destructible usb storage first until you are comfortable, I am comfortable with what I've said above but I do not expect anyone to take me at my word since i'm just some rando on the interwebs. ;)
